packages/graphql-client/src/api-client-utilities/validations.ts

```diff
@@ -1,39 +1,51 @@
 import { APIClientLogger } from "./types";
 
-export { validateRetries } from "../graphql-client/utilities";
-
-export function validateRequiredStoreDomain(
-  errorPrefix: string,
-  storeDomain: string | undefined
-) {
-  if (
-    !storeDomain ||
-    typeof storeDomain !== "string" ||
-    storeDomain.trim().length < 1
-  ) {
-    throw new Error(`${errorPrefix} a valid store domain must be provided`);
+export function validateDomainAndGetStoreUrl({
+  client,
+  storeDomain,
+}: {
+  client: string;
+  storeDomain: string | undefined;
+}) {
+  try {
+    if (!storeDomain || typeof storeDomain !== "string") {
+      throw new Error();
+    }
+
+    const trimmedDomain = storeDomain.trim();
+
+    const url = trimmedDomain.startsWith("http")
```

Comment:
I don't think we need this anymore.

Reply:
We need it because in order for us to use a non protocol domain string (eg. `test.shopify.io` as the provided store domain), I need to check whether the provided domain has the protocol and if not, add it so that we can use the `url` API. The `url` API requires the protocol or else it throws an error.

Reply:
Totally missed that. Nice catch!